### ISSUE TITLE
Function keys on Hacker's Keyboard don't work

### ISSUE THREAD

**Issue by indvd00m** [connectbot/connectbot#137]
Numeric digits are entered instead.

https://code.google.com/p/connectbot/issues/detail?id=693

**Comment by jklein24** [comment]
The function keys on hacker keyboard seem to work fine for me on Master, although I can confirm that the current version on the play store isn't working. @kruton, when are you planning on pushing a new version?

**Comment by kruton** [comment]
I just pushed 1.8.5 to production.

**Comment by jklein24** [comment]
Awesome! @indvd00m, let us know if you still have this problem after the update.

**Comment by indvd00m** [comment]
@jklein24, 1.8.5 - still don't work.

**Comment by leonerd** [comment]
You should be aware that Hackers' Keyboard does a *lot* of very weird things internally when it knows it's talking to ConnectBot, by sending ESC-prefixed escape sequences, rather than real keypresses. This I believe was to work around a bunch of behaviour where CB didn't handle the keys correctly, but since it now does, that logic should be removed.

**Comment by jklein24** [comment]
Hmm it seems that this still works fine building from HEAD, but not in production. The diff between those 2 is pretty small. @kruton is there something in the build/release process that might be causing this?

**Comment by kruton** [comment]
Maybe Hacker's Keyboard has a switch internally based on ConnectBot version numbers.

**Comment by CMDRZOD** [comment]
Bump. I also have this issue. It makes byobu impossible to use on the phone. Please fix it soon. Or contact the makers of hackers keyboard if it's their fault. This has been ongoing since this summer.

**Comment by leonerd** [comment]
@CMDRZOD It's been going on longer than that.
I've repeatedly tried to contact the Hackers' Keyboard maintainers and have repeatedly not got any reply. If anyone knows any way of doing that, please attempt to do so. There's ultimately a sharp limit on what we can fix within ConnectBot without their help fixing their side.

**Comment by leonerd** [comment]
@kruton Yes - HK does do some really weird behavioural switching based on name/version detecting of the client application it is sending keypresses to. I believe that may account for version-related differences between dev builds and production releases.

**Comment by grizzlyfred** [comment]
Yes please fix the f keys to allo usimg midnught commanee

**Comment by grizzlyfred** [comment]
I wish to add that it wirks in vx connectbot... but vxcb doesnt work for all my machines...

**Comment by Gerard987789** [comment]
Try the following: press first Esc and than the desired function key.
For me it works.

Using HK version 1.37 and Connectbot 1.8.6.
